cidc_api/shared/gcloud_client.py

```diff
@@ -213,15 +251,42 @@ def _get_ttl_expression(days_until_expiry: int = INACTIVE_USER_DAYS) -> str:
     return ttl_expression
 
 
-def _is_matching_conditional_binding(
-    binding: storage.bucket.Policy, member_id: str, subexpression: str
-) -> bool:
-    try:
-        has_user = binding["members"] == {member_id}
-        has_condition = subexpression in binding["condition"]["expression"]
-        return has_user and has_condition
-    except KeyError:
-        return False
+def _get_user_download_binding(
```

Comment:
nit: rather then returning an index of a binding (later used to either overwrite or to delete the binding by index)
this fn could been something like `find_and_pop_user_download_binding`, so it will already remove the binding from the policy and overwrite would be as easy as appending new, proper binding to the policy. I can't shake off a (somewhat groundless) feeling that substitution by index is fragile or something.

Reply:
This would also allow to split this function in two - `pop` and `create_empty` - and to call the later in lazy way:
```
   binding =  find_and_pop_user_download_binding() or create_empty_download_binding()
```